src/algorithms/arc_swap.rs

```diff
@@ -272,7 +310,8 @@
     Self: std::fmt::Debug + Copy + PartialOrd + Send + Sync + num::Zero,
     Self: std::iter::Sum + num::FromPrimitive + num::ToPrimitive,
     Self: std::ops::AddAssign + std::ops::SubAssign + std::ops::Sub<Output = Self>,
-    Self: std::ops::Div<Output = Self>,
+    Self: std::ops::Mul<Output = Self> + std::ops::Div<Output = Self>,
```

Comment:
It is unfortunate that we require this far into coupe. To me, it should be more logical to only have max_weights for each part as an entry.

Reply:
`max_part_weight` is computed from either the `max_imbalance` input, or from the input partition (the weight of the overweight part) otherwise.

We can totally add another input to control its value, though I would like to postpone this for another PR.